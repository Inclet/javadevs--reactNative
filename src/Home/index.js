import React, { Component } from "react";
import { View, Text, StyleSheet, Image } from 'react-native';
import commonStyle from "../Styles/common.style";
import themeStyle from "../Styles/theme.style";
import anImage from "../Assets/Images/webdesign.png";
class Home extends Component {
  state = {};

  render() {
    return (
      <View style={commonStyle.LeftAlignLayout}>
        <Text style={styles.pageTitleStyle}>Top Java Developers</Text>
        <View style={styles.developerCardStyle}>
          <Text>01</Text>
          <Image source={anImage} style={styles.profileImageStyle} />
        </View>
      </View>
    );
  }
}

export default Home;

const styles = StyleSheet.create({
  pageTitleStyle: {
    fontSize: 20,
    marginLeft: 10,
    color: themeStyle.PRIMARY_COLOR,
  },
  profileImageStyle: {
    width: 61,
    height: 66
  },
  developerCardStyle: {
    marginLeft: 10,
    flexDirection: 'row',
    alignItems: 'flex-start',
    justifyContent: 'flex-start',
  }
});
