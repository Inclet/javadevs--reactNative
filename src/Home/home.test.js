import React from "react";
import { shallow } from "enzyme";
import { MockedProvider } from '@apollo/react-testing';
import Home from './index';
import javadeveloper from '../graphQL_Queries/getAllJavadevelopers';

const mocks = [
  {
    request: {
      query: javadeveloper,
      variables: {
        name: 'me',
      },
    },
    result: {
      data: {
        developer: { id: 1, name: 'users' }
      }
    }
  },
];
describe("Home Screen", () => {
  const wrapper = shallow(
    <MockedProvider mocks={mocks} addTypename={false}>
      <Home />
    </MockedProvider>
  );

  it("Should match Snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Should show loader initially', () => {
    const component = shallow(<Home />);
  });
});
