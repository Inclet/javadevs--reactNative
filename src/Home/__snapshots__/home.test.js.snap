// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Screen Should match Snapshot 1`] = `ShallowWrapper {}`;
